### DISCUSSION TITLE
Help with 1.3inch LCD HAT (waveshare)

### DISCUSSION THREAD

**Discussion by lordpysc** [jayofelony/pwnagotchi#171] [Q&A]
Hi everyone, I bought the 1.3inch LCD HAT from waveshare. I just need to confirm if the screen is supported. I was looking thru the utils and found (  elif config['ui']['display']['type'] in ('wavesharelcd1in3', 'wslcd1in3'):
        config['ui']['display']['type'] = 'wavesharelcd1in3' ) just need to be sure if thats the one to go to. Thanks in advance,

**Comment by jayofelony** [comment]
Yep should be it, let me know if it works

**Reply by lordpysc** [reply]
> Yep should be it, let me know if it works

Thanks Jay, your work is awesome. Can I switch between screens? Like going to config.toml and change the waveshare_4 to the new one? Or so I have to re flash? 

I was supposed to get 2 rbpi2w but the vilros package came without it so going back and forth with Amazon. Ended up with one for the moment, thanks in advance

**Reply by jayofelony** [reply]
Yeah its interchangeable, just disable the display in config, turn the pi off and pop another display on. Ad after that you can turn it on again and edit the config accordingly.

**Reply by lordpysc** [reply]
> Yeah its interchangeable, just disable the display in config, turn the pi off and pop another display on. Ad after that you can turn it on again and edit the config accordingly.

Thank you so much, doing it now. I will post a pic if it works :D

**Reply by lordpysc** [reply]
> Yeah its interchangeable, just disable the display in config, turn the pi off and pop another display on. Ad after that you can turn it on again and edit the config accordingly.

I did it, but screen is black, it changes in webui but on my screen I can see the backlight on but everything is dark, I tried changing the ui.invert = false just in case but same result.

 ![image](https://github.com/jayofelony/pwnagotchi/assets/145384942/b859d443-9d6e-45c0-bd50-5c56e2745728)![image](https://github.com/jayofelony/pwnagotchi/assets/145384942/8107048d-aa34-477b-9bfe-4987c4c0f2f9)

**Reply by jayofelony** [reply]
try this:
0W:
`sudo nano /boot/config.txt` find the spi0-0cs in your `[pix]` section and change it to spi0-2cs.

02W, 3, 4 and 5:
`sudo nano /boot/firmware/config.txt` find the spi0-0cs in your `[pix]` section and change it to spi0-2cs.

---
`ui.invert = true` makes the the background white

**Reply by lordpysc** [reply]
Ok will do, thanks so much for taking the time to point me in the right direction, kudos for all the hard work you do!

**Reply by lordpysc** [reply]
> try this:
> 
> 0W:
> 
> `sudo nano /boot/config.txt` find the spi0-0cs in your `[pix]` section and change it to spi0-2cs.
> 
> 
> 
> 02W, 3, 4 and 5:
> 
> `sudo nano /boot/firmware/config.txt` find the spi0-0cs in your `[pix]` section and change it to spi0-2cs.
> 
> 
> 
> ---
> 
> `ui.invert = true` makes the the background white

I did it, and saw my screen flashing white for a second then went back to backlight black. 

I went ahead and did a fresh flash on a Rb Pi 4, selected the screen, but same black backlight and I also changed the spi0-2cs, same one flash white and back to black. 

![image](https://github.com/jayofelony/pwnagotchi/assets/145384942/8ea4273a-8fda-42e2-99d6-1bc8433a1011)

**Reply by lordpysc** [reply]
> > try this:
> > 
> > 0W:
> > 
> > `sudo nano /boot/config.txt` find the spi0-0cs in your `[pix]` section and change it to spi0-2cs.
> > 
> > 
> > 
> > 02W, 3, 4 and 5:
> > 
> > `sudo nano /boot/firmware/config.txt` find the spi0-0cs in your `[pix]` section and change it to spi0-2cs.
> > 
> > 
> > 
> > ---
> > 
> > `ui.invert = true` makes the the background white
> 
> I did it, and saw my screen flashing white for a second then went back to backlight black. 
> 
> I went ahead and did a fresh flash on a Rb Pi 4, selected the screen, but same black backlight and I also changed the spi0-2cs, same one flash white and back to black. 
> 
> ![image](https://github.com/jayofelony/pwnagotchi/assets/145384942/8ea4273a-8fda-42e2-99d6-1bc8433a1011)

My friend Jay, I was able to find a fix for it. After a few hours of research, I was able to figure the type of screen 'lcdhat' and removed rotation cuz for some reason with rotation even set to 0 screen was unresponsive.

Now trying to find a fix for full screen lol

Proud dad of 2, soon to be 3 gotchis 😂
Thanks for your support.
![image](https://github.com/jayofelony/pwnagotchi/assets/145384942/954483bc-8085-4dcf-9bb1-b3b3b5ee4052)

**Reply by jayofelony** [reply]
Try switching the sizes around if that works
